code/game/machinery/_machinery.dm

```diff
@@ -433,8 +471,12 @@ Class Procs:
 		occupant = null
 		update_icon()
 		updateUsrDialog()
+		return ..()
 	if(A == circuit)
 		circuit = null
+	if(A in component_parts)
+		component_parts.Remove(A)
```

Comment:
should be safe to skip this if src is being qdeleted

Reply:
Are you suggesting an early return for the entire proc `if(QDELETED(src))`?

Reply:
@ShizCalev Still waiting on this. Are you saying - handle_atom_del is only called when an object is qdeleted, so it should be safe to not call the parent proc, component_parts.Remove() and RefreshParts()?

Reply:
handle_atom_del is called when an atom in the parents's contents is deleted.

however Destroy() QDEL_LIST's everything in component parts, which means that if something has 8 component parts, it's going to call RefreshParts() 8 times during a single Destroy() call, which is essentially just wasted processing since the thing is getting deleted anyway.

can just change it to if((A in component_parts) && !QDELETED(src))